### ISSUE TITLE
Suggestions: Link to a build process and simplify configuration

### ISSUE THREAD

**Issue by liammclennan** [codereflection/Giles#2]
These aren't issues - more like some ideas.

Giles makes a couple of big assumptions: nunit and msbuild. Any self-respecting application is going to have some kind of build process in place so it makes sense to connect Giles to that. Rather than calling msbuild and nunit Giles should be able to execute commands and, if you feel generous, possibly provide convenient configuration for the common build tools (msbuild, nant, rake, psake). Also, if you remove the hard-coded reference to .cs files then there is no reason why Giles is restricted to .NET. 

I forked Giles and experimented with these ideas, but I made a horrible mess. :)

**Comment by codereflection** [comment]
These ideas are great! You're right, Giles does make some big assumptions, which was on purpose. While I have no intention of limiting Giles to NUnit and MSpec, I did make a conscious to not run a build script. The sole reason being speed of execution. What I wanted out of Giles was immediate build and test results, as fast as possible.  Using the console runners themselves is either going to go away completely, or be an option to switch back to.  I'm currently working on a plugin structure for Giles that will allow it to run the test frameworks directly, sans console runner, giving the dev more granular control over exactly what Giles is running for the tests.  The first iteration will be support for mspec and specific namespaces and members.

Having said that, giving Giles the ability to run build tools like those you mentioned is a great idea. I'm thinking that given the interactive console, switching between running just a build/test and a full build script could be very simple, and extremely useful.  We could even have Giles pass a custom list of tasks to the build scripts. Yep, love the idea.

I do eventually want to make the watched file types configurable, and I even felt dirty hard coding "*.cs".. and will be forced to do so when Giles supports F#, IronRuby, IronPython, etc.

I'd love to checkout your fork when I have a few minutes later today.

**Comment by liammclennan** [comment]
Great point re: speed. I want Giles to call rake, and that is always going to be damn slow, but it adds a lot of flexibility. I started using Giles and found that I wanted it to just run my fast tests (Category("Fast")) and therefore need a way to pass a category to nunit-console. But then I already have Rake tasks for fast and slow tests so I thought that might be easier. 

There is not much to check out in my fork - I stopped working on it when I had completely ruined it.
